flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/OperationTreeBuilder.scala

```diff
@@ -0,0 +1,317 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.plan
+
+import java.util.{Optional, List => JList, Map => JMap}
+
+import org.apache.flink.api.java.operators.join.JoinType
+import org.apache.flink.table.api._
+import org.apache.flink.table.expressions.{Alias, Asc, Expression, Ordering,
+  PlannerExpressionConverter, ResolvedFieldReference, UnresolvedAlias, WindowProperty}
+import org.apache.flink.table.functions.utils.UserDefinedFunctionUtils
+import org.apache.flink.table.functions.{TemporalTableFunction, TemporalTableFunctionImpl}
+import org.apache.flink.table.plan.ProjectionTranslator.{expandProjectList, flattenExpression,
+  resolveOverWindows}
+import org.apache.flink.table.plan.logical._
+import org.apache.flink.table.util.JavaScalaConversionUtil.toScala
+
+import _root_.scala.collection.JavaConverters._
+
+/**
+  * Builder for [[[Operation]] tree.
+  */
+class OperationTreeBuilder(private val tableEnv: TableEnvironment) {
+
+  private val expressionBridge: PlannerExpressionConverter = PlannerExpressionConverter.INSTANCE
```

Comment:
This makes it impossible for the Blink people to plugin their own converter. Why not using the field of `tableEnv`.